python/IECore/Enum.py

```diff
@@ -50,11 +52,14 @@
 #
 # assert( E.Orange == E( 1 ) )
 # assert( E.Orange != E.Apple )
+
+class EnumBase( object ): pass
```

Comment:
This fixes the comparision issues since they now have a common base class and `type()` works as before.

Reply:
I don't understand this. I've tested here, and I can't reproduce the `gaffer browser` startup error you described on the other PR, with or without the use of `EnumBase`. But I don't see how EnumBase can have solved the problem for you either, because the types for two different enums are still different, even if they have a common base class. I tested here and this is what I get:

```
A = IECore.Enum.create( "X", "Y" )
B = IECore.Enum.create( "U", "V" )
print type( A.X ), type( B.U ), type( A.X ) is type( B.U )
print A.X == B.U

# Output from this PR :

<class 'IECore.Enum.Enum'> <class 'IECore.Enum.Enum'> False
Traceback (most recent call last):
  File "t.py", line 7, in <module>
    print A.X == B.U
  File "/disk1/john/dev/build/gaffer/python/IECore/Enum.py", line 89, in __eq__
    assert( type( self ) is type( other ) )
AssertionError

# Output from version without EnumBase, but deriving from object

<class 'IECore.Enum.Enum'> <class 'IECore.Enum.Enum'> False
Traceback (most recent call last):
  File "t.py", line 7, in <module>
    print A.X == B.U
  File "/disk1/john/dev/build/gaffer/python/IECore/Enum.py", line 89, in __eq__
    assert( type( self ) is type( other ) )
AssertionError

# Output from original code (no base class) :

<type 'instance'> <type 'instance'> True
True
```

I suspect the Gaffer error could be coming from custom IE code that mixes `ListContainer.Orientation` with `SplitContainer.Orientation`. I should probably combine those two into one enum in Gaffer.

Ultimately, none of this is probably very important, but I want to understand why I'm not seeing what you describe with EnumBase etc...

Reply:
Checking it again, with the latest Cortex environment it also fails now on my end. I was also a bit confused how that base class fixed that issue and now I suspect I had some mixed up environments in my test. Sorry about that. 

```
File "/software/apps/gaffer/0.54.2.3/cent7.x86_64/cortex/10/gaffer/python/GafferCortexUI/FileIndexedIOPathPreview.py", line 53, in __init__
    with GafferUI.SplitContainer( GafferUI.ListContainer.Orientation.Horizontal ) :
  File "/software/apps/gaffer/0.54.2.3/cent7.x86_64/cortex/10/gaffer/python/GafferUI/SplitContainer.py", line 59, in __init__
    self.setOrientation( orientation )
  File "/software/apps/gaffer/0.54.2.3/cent7.x86_64/cortex/10/gaffer/python/GafferUI/SplitContainer.py", line 71, in setOrientation
    v = QtCore.Qt.Horizontal if orientation == self.Orientation.Horizontal else QtCore.Qt.Vertical
  File "/home/timle/apps/cortex/10.0.0-a70dev/cent7.x86_64/base/python/2.7/gcc/4.8.3/IECore/Enum.py", line 87, in __eq__
    assert( type( self ) is type( other ) )
AssertionError
```

The issue really seems to be in `GafferCortexUI` comparing two different enums. So I'm gonna fix that on 0.54 maintenance and go back to `object` as baseclass for Enum.

Reply:
Fix: https://github.com/GafferHQ/gaffer/pull/3558

Reply:
Thanks - I'm glad we've got to the bottom of this. LGTM.